libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1726,6 +1727,19 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 
 		break;
 	}
+	case FunctionType::Kind::ERC7201:
+	{
+		solAssert(arguments.size() == 1);
+		auto typedRational = ConstantEvaluator::tryEvaluate(_functionCall);
+		solAssert(typedRational && std::holds_alternative<rational>(typedRational->value));
+		auto rationalValue = std::get<rational>(typedRational->value);
+		solAssert(rationalValue.denominator() == 1);
+		bigint value = rationalValue.numerator();
+		solAssert(value <= std::numeric_limits<u256>::max());
+
+		define(_functionCall) << formatNumber(u256(value)) << "\n";
+		break;
+	}
```

Comment:
You need actual runtime implementation here. If we reached this place, it means we encountered the builtin outside the compile-time context (i.e. layout specifier or array size). In that case we just insert the runtime implementation and rely on the optimizer to potentially reduce it to something simpler. That's supposed to happen even if the argument is a literal.